File: backend/src/routes/agents.js
/**
 * GET  /agents        — aggregated status of all AI agents
 * POST /agents/start  — start the fleet (idempotent)
 * POST /agents/stop   — stop the fleet
 * GET  /agents/metrics — live metrics snapshot (pricing, monitoring, revenue)
 */
const express = require('express');
const router = express.Router();
const orchestrator = require('../agents/AgentOrchestrator');

// GET /agents
router.get('/', (_req, res) => {
  res.json(orchestrator.status());
});

// GET /agents/metrics
router.get('/metrics', (_req, res) => {
  const full = orchestrator.status();
  res.json(full.metrics);
});

// POST /agents/start
router.post('/start', (_req, res) => {
  orchestrator.start();
  res.json({ message: 'Agent fleet started', status: orchestrator.status() });
});

// POST /agents/stop
router.post('/stop', (_req, res) => {
  orchestrator.stop();
  res.json({ message: 'Agent fleet stopped' });
});

module.exports = router;

File: fastapi_backend/app/routers/agents.py
from fastapi import APIRouter
from app.agents.orchestrator import orchestrator

router = APIRouter()


@router.get("/")
async def get_agents():
    return orchestrator.status()


@router.get("/metrics")
async def get_metrics():
    return orchestrator.status().get("metrics", {})


@router.post("/start")
async def start_agents():
    orchestrator.start()
    return {"message": "Agent fleet started", "status": orchestrator.status()}


@router.post("/stop")
async def stop_agents():
    orchestrator.stop()
    return {"message": "Agent fleet stopped"}
